docs/portable-modules/portable-modules.md

```diff
@@ -0,0 +1,54 @@
+Other facade assemblies are not shipping in Windows and if new facade assemblies are included, they will only ship in newer versions of Windows.
+
+## System.Management.Automation
+
+We need to produce a set of reference assemblies for `System.Management.Automation.dll`.
+In the past, we have published the asmmeta produced reference assemblies because it was simple and expedient, but that will not work going forward.
+Note that we shipped multiple reference assemblies (e.g. for `Microsoft.PowerShell.Commands.Utility.dll`) for unclear reasons - many of those assemblies do not contain apis that we really think of as a supported api despite the classes being public.
+
+Current guidance is to use the PowerShell 6 reference assemblies because those target `netstandard1.6`.
+This is also not an option because modules may take an accidental dependency on apis not present in older version of PowerShell.
+
+The correct solution is to create a new reference assembly that contains only the apis that are truly portable.
+To that end, I started with the V3 version of `System.Management.Automation.dll` and generated C# for the public api surface using an internal tool called `GenApi`.
+I then created a project with `dotnet new`, set the target framework to `netstandard1.3`, and removed apis that didn't compile because there is no equivalent api in .Net Core.
+
+To target newer versions of PowerShell, we could take this C#, add the new apis under `#if` so that we have one simple definition of our public api.
+We could also take the opportunity to revisit many of the apis - think of this as our way of limiting the public api surface without removing the code.
+
+## Practical Experience
+
+I have successfully built `PSReadline` targeting `netstandard1.3` and the `System.Management.Automation.dll` reference assembly mentioned above.
+I needed to binplace 7 facade assemblies to get `PSReadline` to load and work correctly in Windows PowerShell.
+These facade assemblies are needed for types like `System.Collections.Hashtable`, `System.Console`, `System.Threading.Thread`, and `System.IO.File`.
+
+The single binary worked correctly in Windows PowerShell with the facade assemblies and in PowerShell Core without the facade assemblies.
+
+## Solutions
+
+I can see several possible solutions, in roughly decreasing order of preference:
+
+* `PowerShellGet` could detect read the metadata from binary modules it is installing and automatically install the facade assemblies that are required.
+  Installation should ideally go in a the GAC or alongside `powershell.exe`, but that requires elevation, so installing along with the module is an option.
+
+* We can chose a set of facade assemblies to install on older versions of Windows.
+  These assemblies can be installed in either the GAC or alongside `powershell.exe`.
+
+* We can require module authors to package the facade assemblies with their module.
```

Comment:
This is a general problem for anyone developing against .Net Standard using CoreClr and wanting to deploy onto FullClr.  Is there any guidance from dotnet?